programs/manifest/src/program/processor/shared.rs

```diff
@@ -0,0 +1,146 @@
+use std::{
+    cell::{Ref, RefMut},
+    mem::size_of,
+};
+
+use crate::{
+    state::{constants::BLOCK_SIZE, DynamicAccount, GlobalFixed, MarketFixed},
+    validation::{ManifestAccount, ManifestAccountInfo, Program, Signer},
+};
+use bytemuck::Pod;
+use hypertree::{get_helper, get_mut_helper, trace};
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult, program::invoke, rent::Rent,
+    system_instruction, sysvar::Sysvar,
+};
+
+pub(crate) fn expand_market_if_needed<'a, 'info, T: ManifestAccount + Pod + Clone>(
+    payer: &Signer<'a, 'info>,
+    manifest_account: &ManifestAccountInfo<'a, 'info, T>,
+    system_program: &Program<'a, 'info>,
+) -> ProgramResult {
+    let need_expand: bool = {
+        let market_data: &Ref<&mut [u8]> = &manifest_account.try_borrow_data()?;
+        let fixed: &MarketFixed = get_helper::<MarketFixed>(market_data, 0_u32);
+        !fixed.has_free_block()
+    };
+
+    if !need_expand {
+        return Ok(());
+    }
+    expand_dynamic(payer, manifest_account, system_program)?;
+    expand_market_fixed(manifest_account.info)?;
+    Ok(())
+}
+
+// Expand is always needed because global doesnt free bytes ever.
```

Comment:
Correct. Similar to seats on market, no need to shrink. This could in theory be a problem if someone spammed and filled out a global account. We can reconsider if needed